lib/LongVectorLoweringPass.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 The Clspv Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "llvm/Pass.h"
+#include "llvm/Support/Debug.h"
+
+#include "clspv/Passes.h"
+
+using namespace llvm;
+
+#define DEBUG_TYPE "LongVectorLowering"
+
+namespace {
+
+class LongVectorLoweringPass final : public ModulePass {
+public:
+  static char ID;
+
+public:
+  LongVectorLoweringPass() : ModulePass(ID) {}
+
+  /// Lower the content of the given module @p M.
+  bool runOnModule(Module &M) override;
+};
+
+char LongVectorLoweringPass::ID = 0;
+
+bool LongVectorLoweringPass::runOnModule(Module &) {
+  // TODO implement long-vector lowering.
+  bool Modified = false;
+  return Modified;
+}
+
+} // namespace
+
+static RegisterPass<LongVectorLoweringPass>
```

Comment:
Please use the initialization macro as in the other files (adding to passes.{cpp,h}). I've changed all other passes to do that and is easier to work with for some internal builds.

Reply:
The downside of this macro is that we cannot use `--print-before/after=LongVectorLowering`. Are you aware of a workaround that I can use with the macro?

Reply:
I usually just dump everything into a file and scan for the particular pass.

Reply:
I see. I've updated the code to use the macro.